pkg/gatherers/clusterconfig/gather_openstack_helpers.go

```diff
@@ -0,0 +1,94 @@
+package clusterconfig
+
+import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/anonymization"
+	"github.com/openshift/insights-operator/pkg/utils/anonymize"
+)
+
+// This function removes from the CR which is represented by the map[string]interface{} data,
+// fields listed in the 'fieldsToRemove' list. Each field has to have full path provided to be removed.
+func removeFields(data map[string]interface{}, fieldsToRemove [][]string) map[string]interface{} {
+	for _, fieldToRemove := range fieldsToRemove {
+		unstructured.RemoveNestedField(data, fieldToRemove...)
+	}
+	return data
+}
+
+// This function goes over all fields in the provided map and anonymize all IPv4 addresses which will find
+func anonymizeIpAddresses(data map[string]interface{}) map[string]interface{} {
+	ipv4Regex := regexp.MustCompile(anonymization.Ipv4Regex)
+	for fieldName, fieldValue := range data {
+		switch currentValue := fieldValue.(type) {
+		case map[string]interface{}:
+			data[fieldName] = anonymizeIpAddresses(currentValue)
+		default:
+			currentValueStr, _ := currentValue.(string)
+			isIpv4 := ipv4Regex.FindStringIndex(currentValueStr)
+			if isIpv4 != nil {
+				data[fieldName] = anonymize.String(currentValueStr)
+			}
+		}
+	}
+	return data
```

Comment:
We anonymize all the IP addresss as opt-in feature so I would not go the extra mile here.

Reply:
so should I check value of the `enableGlobalObfuscation` option to do or don't do this anonymization? Or I missed something there?

Reply:
you don't need to do anything. If the flag is enabled then all the IP addresses are obfuscated before writing the data to an archive.

Reply:
ahh, ok. I didn't know that :) Thx a lot